app/components/Calendar.tsx

```diff
@@ -10,33 +14,65 @@ const initialEventsList: CalEvent[] = [
     title: 'timed event',
     start: moment().add(1, 'hour').toDate(),
     end: moment().add(2, 'hours').toDate(),
+    description: '',
+    location: '',
+    visibility: 'private',
+    attendees: ['a@b.c', 'alpha@beta.gamma'],
   },
   {
     id: '2',
     title: 'Some title',
     start: moment().add(1, 'day').toDate(),
     end: moment().add(1, 'day').add(2, 'hour').toDate(),
+    description: '',
+    location: '',
+    visibility: 'public',
+    attendees: ['alpha@beta.gamma', 'admin@rds.com'],
   },
   {
     id: '3',
     title: 'Another title',
     start: moment().add(1, 'day').add(8, 'hours').toDate(),
     end: moment().add(1, 'day').add(9, 'hours').toDate(),
+    description: '',
+    location: '',
+    visibility: 'private',
+    attendees: ['admin@rds.com'],
   },
 ];
 
-const Calendar = () => {
+interface CalendarProps {
+  view?: View;
+}
+
+const Calendar = ({ view }: CalendarProps) => {
   const [eventsList, setEventsList] = useState<CalEvent[]>(initialEventsList);
-  const [calendarEvent, setCalendarEvent] = useState<CalendarEventProps>();
+  const [calendarEvent, setCalendarEvent] = useState<CalendarEventProps>({
+    event: initialEventsList[0],
+    show: false,
+    new: false,
+  });
 
-  const updateEventState = ({ event, start, end }: UpdateEvent) => {
+  const updateEventState = (event: CalEvent) => {
     setCalendarEvent((e) => ({ ...e, event }));
     setEventsList((events) =>
       events.map((e) => {
         if (e.id === event.id) {
           e.title = event.title;
-          e.start = moment(start).toDate();
-          e.end = moment(end).toDate();
+          e.start = moment(event.start).toDate();
+          e.end = moment(event.end).toDate();
+        }
+        return e;
+      })
+    );
+  };
+
+  const updateEventStateFromModal = (event: CalEvent) => {
+    setCalendarEvent((e) => ({ ...e, event }));
+    setEventsList((events) =>
+      events.map((e) => {
```

Comment:
```suggestion
      events.map(({ id }) => {
```
NIT

Reply:
@sumitd94 we will need it as we are returning `e` on line 77.

Reply:
Resolving the issue as it is needed